components/Controllers/SwapPositionsPageContent/SwapPositionsPageContent.tsx

```diff
@@ -68,82 +69,206 @@ export function SwapPositionsPageContent() {
+          <span>=</span>
         </div>
-        <div className={`col5 ${styles.column} `}>
-          <br />
-          <br />
-          <br />
-          <br />
-          <br />
-          <br />
-          <br />
-          <div className={styles.yellow}>
+        <div className={styles.bottomMath}>
+          <span className={styles.yellow}>
             total unrealized gain/loss:{' '}
             {(exitValue - (averagePurchased - averageSold)).toFixed(4)} ETH (
             {(exitValue / (averagePurchased - averageSold) - 1).toFixed(4)} %)
-          </div>
+          </span>
         </div>
       </div>
+
+      <br />
+      <p className={styles.diffExplainer}>
+        {' '}
+        The below &rdquo;running total papr bought minus sold&rdquo; might
+        differ from &rdquo;papr bought - papr sold&rdquo; shown above because
+        above we model as if you exited all LP positions this moment. Accounting
+        will also be slightly broken if the EOA has used paprMEME individually
+        and as a signer on a multisig.
+      </p>
+      {addressToUse && addressToUse != '' && (
+        <ActivityTimeline
+          account={addressToUse}
+          startTime={timestamps.start}
+          endTime={timestamps.end}
+        />
+      )}
     </>
   );
 }
+
+interface ActivityTimelineProps {
+  account: string;
+  startTime: number;
+  endTime: number;
+}
+
+import { ActivityType } from 'hooks/useActivity/useActivity';
+import { useLPActivityAndImplicitSwaps } from 'hooks/useLPActivityAndImplicitSwaps';
+import { usePoolTokens } from 'hooks/usePoolTokens';
+
+import { getActivityKind } from '../Activity/Activity';
+
+interface ActivityWithRunningBalance extends ActivityType {
+  runningPaprDebt?: number;
+  runningPaprBalance?: number;
+  paprDelta?: number;
+  ethDelta?: number;
+  kind?: string;
+}
+
+const ActivityTimeline: React.FC<ActivityTimelineProps> = ({
+  account,
+  startTime,
+  endTime,
+}) => {
+  const {
+    data: activityData,
+    fetching: activityFetching,
+    fetchMore,
+    remaining,
+  } = useActivity(
+    '0x3b29c19ff2fcea0ff98d0ef5b184354d74ea74b0',
+    account,
+    undefined,
+    500,
+  );
+
+  const { implicitSwaps } = useLPActivityAndImplicitSwaps(
+    account,
+    startTime,
+    endTime,
+  );
+
+  const { token0, token1 } = usePoolTokens();
+
+  const [activities, setActivities] = useState<
+    ActivityWithRunningBalance[] | null
+  >(null);
+
+  useEffect(() => {
+    const filteredActivities = activityData.filter(
+      (a) => a.timestamp >= startTime && a.timestamp <= endTime,
+    );
+    filteredActivities.reverse();
+    let runningTotal = 0;
+    let runningDebtTotal = 0;
+    const a: ActivityWithRunningBalance[] = filteredActivities.map((d) => {
+      runningTotal += d.tokenIn
+        ? d.tokenIn.symbol == 'paprMEME'
+          ? (d.amountIn / 1e18) * -1
+          : d.amountOut / 1e18
+        : 0;
+      runningDebtTotal += d.amountBorrowed ? d.amountBorrowed / 1e18 : 0;
```

Comment:
notice you've been directly dividing by 1e18, we should use parseFloat + formatUnits + token decimals to be consistent

Reply:
papr tokens are always 18 decimals?

Reply:
I'll clean up underlying to make it more generalized, but I assume we'd refactor all of this if we wanted to keep it